tasks/create-and-push-cib.yml

```diff
@@ -75,34 +75,114 @@
+        resource: "{{ item }}"
+        resource_is_stonith: "{{ item.agent.startswith('stonith:') }}"
+      loop: "{{ ha_cluster_resource_primitives }}"
+
+    - name: Configure cluster resource groups
+      include_tasks: pcs-cib-resource-group.yml
+      vars:
+        resource_group: "{{ item }}"
+      loop: "{{ ha_cluster_resource_groups }}"
+
+    - name: Configure cluster resource clones
+      include_tasks: pcs-cib-resource-clone.yml
+      vars:
+        resource_clone: "{{ item }}"
+      loop: "{{ ha_cluster_resource_clones }}"
+
+    ## Constraints
+
+    - name: Configure resource location constraints
+      include_tasks: pcs-cib-constraint-location.yml
+      loop: "{{ ha_cluster_constraints_location }}"
+      loop_control:
+        index_var: constraint_index
+        loop_var: constraint
+
+    - name: Configure resource colocation constraints
+      include_tasks: pcs-cib-constraint-colocation.yml
+      when: not constraint.resource_sets | default()
+      loop: "{{ ha_cluster_constraints_colocation }}"
+      loop_control:
+        index_var: constraint_index
+        loop_var: constraint
+
+    - name: Configure resource set colocation constraints
+      include_tasks: pcs-cib-constraint-set.yml
+      vars:
+        constraint_type: colocation
+      when: constraint.resource_sets | default()
+      loop: "{{ ha_cluster_constraints_colocation }}"
+      loop_control:
+        index_var: constraint_index
+        loop_var: constraint
+
+    - name: Configure resource order constraints
+      include_tasks: pcs-cib-constraint-order.yml
+      when: not constraint.resource_sets | default()
+      loop: "{{ ha_cluster_constraints_order }}"
+      loop_control:
+        index_var: constraint_index
+        loop_var: constraint
+
+    - name: Configure resource set order constraints
+      include_tasks: pcs-cib-constraint-set.yml
+      vars:
+        constraint_type: order
+      when: constraint.resource_sets | default()
+      loop: "{{ ha_cluster_constraints_order }}"
+      loop_control:
+        index_var: constraint_index
+        loop_var: constraint
+
+    - name: Configure resource ticket constraints
+      include_tasks: pcs-cib-constraint-ticket.yml
+      when: not constraint.resource_sets | default()
+      loop: "{{ ha_cluster_constraints_ticket }}"
+      loop_control:
+        index_var: constraint_index
+        loop_var: constraint
+
+    - name: Configure resource set ticket constraints
+      include_tasks: pcs-cib-constraint-set.yml
+      vars:
+        constraint_type: ticket
+      when: constraint.resource_sets | default()
+      loop: "{{ ha_cluster_constraints_ticket }}"
+      loop_control:
+        index_var: constraint_index
+        loop_var: constraint
 
-- name: Configure cluster resource clones
-  include_tasks: pcs-cib-resource-clone-pcs-0.10.yml
   vars:
-    resource_clone: "{{ item }}"
-  loop: "{{ ha_cluster_resource_clones }}"
-
+    __ha_cluster_pcs_cli_role:
+      promoted: "{%
```

Comment:
simpler to use `ternary`:
```
promoted: "{{ 'pcmk.cib.roles.promoted-unpromoted' in __ha_cluster_pcs_capabilities |
                     ternary('Promoted', 'Master') }}"
unpromoted: "{{ 'pcmk.cib.roles.promoted-unpromoted' in __ha_cluster_pcs_capabilities |
                     ternary('Unpromoted', 'Slave') }}"
```

Reply:
Nevermind - I see you use `ternary` below - so if you want to keep the code as-is for readability purposes, that's fine with me

Reply:
Ternary actually looks better to me, so I went with that. Thanks for pointing it out.